echopype/calibrate/calibrate_base.py

```diff
@@ -1,19 +1,216 @@
 import abc
+from typing import Dict, List
 
-ENV_PARAMS = ("temperature", "salinity", "pressure", "sound_speed", "sound_absorption")
+import numpy as np
+import scipy.interpolate
+import xarray as xr
+from typing_extensions import Literal
 
 CAL_PARAMS = {
     "EK": ("sa_correction", "gain_correction", "equivalent_beam_angle"),
     "AZFP": ("EL", "DS", "TVR", "VTX", "equivalent_beam_angle", "Sv_offset"),
 }
 
+DataKind = Literal["stationary", "mobile", "organized"]
+InterpMethod = Literal["linear", "nearest", "zero", "slinear", "quadratic", "cubic"]
+ExtrapMethod = Literal["linear", "nearest"]
+VALID_INTERP_METHODS: Dict[DataKind, List[InterpMethod]] = {
+    "stationary": ["linear", "nearest", "zero", "slinear", "quadratic", "cubic"],
+    "mobile": ["linear", "nearest", "cubic"],
+    "organized": ["linear", "nearest"],
+}
+
+
+class EnvParams:
+    def __init__(
+        self,
+        env_params: xr.Dataset,
+        data_kind: DataKind,
+        interp_method: InterpMethod = "linear",
+        extrap_method: ExtrapMethod = "linear",
+    ):
+        """
+        Class to hold and interpolate external environmental data for calibration purposes.
+
+        This class can be used as the `env_params` paremter in `echopype.calibrate.compute_Sv`
+        or `echopype.calibrate.compute_Sp`. It is intended to be used with environmental parameters
+        indexed by time. Environmental parameters will be interpolated onto dimensions within
+        the Platform group of the `EchoData` object being used for calibration.
+
+        Parameters
+        ----------
+        env_params : xr.Dataset
+            The environmental parameters to use for calibration. This data will be interpolated with
+            a provided `EchoData` object.
+
+            When `data_kind` is `"stationary"`, env_params must have a coordinate `"ping_time"`.
+            When `data_kind` is `"mobile"`, env_params must have coordinates `"latitude"`
+            and `"longitude"`.
+            When `data_kind` is `"mobile"`, env_params must have coordinates `"time"`,
+            `"latitude"`, and `"longitude"`.
```

Comment:
Typo: this is the "organized" case?

Reply:
Also it would be good add that the "organized" case is not supported yet -- it would be redundant to what you have below, but many people don't read docs carefully...